float64 dt
float64 endtime
int64 i
